dali/util/device_guard.h

```diff
@@ -30,13 +30,17 @@ class DeviceGuard {
     CUDA_CALL(cudaGetDevice(&original_device_));
```

Comment:
This function never fails.

Reply:
Changed to abort

Reply:
Actually, in some rare cases it can fail. Checking it in ctor seems to be reasonable